### PR TITLE
feat: add updated and created to formMetadata [401088]

### PR DESCRIPTION
**Pull request by KiranVarma02**

### CODE REVIEW

**Review by alexluckett — COMMENTED** [review]
This would benefit from tests to make sure updatedAt/updatedBy is updated when a draft/live is updated too.

File: src/api/forms/service.test.js [hunk 1]
```diff
@@ -310,15 +326,19 @@ describe('Forms service', () => {
...
+      )
+    })
+
     test.each(slugExamples)(`should return slug '$output'`, async (slugIn) => {
       const input = {
         ...formMetadataInput,
         title: slugIn.input
       }
 
-      await expect(updateFormMetadata(id, input)).resolves.toEqual(
```
Comment (KiranVarma02) [inline src/api/forms/service.test.js:319]: It was testing wrong expect here, I corrected it now

File: src/api/forms/service.js [hunk 2]
```diff
@@ -176,7 +176,9 @@ export async function updateDraftFormDefinition(formId, definition, author) {
           {
             $set: {
               'draft.updatedAt': now,
-              'draft.updatedBy': author
+              'draft.updatedBy': author,
+              updatedAt: now,
+              updatedBy: author
```
Comment (colinrotherham) [inline src/api/forms/service.js:181]: Thanks @KiranVarma02 for adding these

Do you mind updating the comment above? It only says we're updated the draft state

```js
// Update the `updatedAt/By` fields of the draft state
```

**Review by davidjamesstone — APPROVED** [review]

**Review by alexluckett — APPROVED** [review]

File: src/api/forms/service.js [hunk 3]
```diff
@@ -216,7 +219,11 @@ export async function updateFormMetadata(formId, formUpdate) {
     }
 
     /** @type {Partial<FormMetadataDocument>} */
-    const updatedForm = { ...formUpdate }
+    const updatedForm = {
+      ...formUpdate,
+      updatedAt: new Date(),
+      updatedBy: author
```
Comment (colinrotherham) [inline src/api/forms/service.js:225]: Should we consolidate `updatedAt` and `updatedBy` into the `formMetadata.update()` method?

i.e. Apply the latest author/date in a single place

**Pull request merged**